### ISSUE TITLE
Predictions are too bad

### ISSUE THREAD

**Issue by mzhadigerov** [thibaultvt/table-transformer-simple-inference#1]
Hi! Thank you for this repo! I tried the pre-trained model and results are too bad (not sure how could the model achieve such a high metrics on test set). Here are some samples:
![4_viz](https://user-images.githubusercontent.com/55549813/169250919-aa0ad397-9377-4e4f-b11d-ecce09764441.jpg)
![5_viz](https://user-images.githubusercontent.com/55549813/169250926-647b0d1d-8465-44be-b3f5-69d001479a55.jpg)
![1_viz](https://user-images.githubusercontent.com/55549813/169250937-ec8e5974-7ccf-4d0a-bb27-5a2d50a7cb8d.jpg)
![2_viz](https://user-images.githubusercontent.com/55549813/169250945-eff7d1c9-ee9f-40d5-ba43-0117424d836b.jpg)
![3_viz](https://user-images.githubusercontent.com/55549813/169250997-018fa561-206a-4a67-b383-f6a561cae4e5.jpg)

Are there any requirements for input images? (e.g.: image size etc.)

**Comment by baulbo** [comment]
@mzhadigerov, this probably has something to do with my custom post processing + visualization rather than with the pre-trained model. Padding needs to be applied on the images before processing, but I added this to the inference code so this shouldn't be the problem.

Try moving the call to the visualization function:
```python
if debug:
            visualization = visualize_structure(image, objs)
            out_path = "./visualization.jpg"
            cv2.imwrite(out_path, visualization)
            logging.info(f"Visualization can be found at '{out_path}'.")

```

right after this line:

`objs = predictions_to_objects(results, threshold, get_class_map(key="index"))`

This should visualize all the objects (not only columns and rows which I filtered out for a personal project). To see of which type the objects can be, inspect the get_class_map function.

```json
{
            'table': 0,
            'table column': 1,
            'table row': 2,
            'table column header': 3,
            'table projected row header': 4,
            'table spanning cell': 5,
            'no object': 
}
```

You could also try increasing the score threshold, e.g., 
`def predict(self, image_path, debug=True, threshold=0.85):`

The post-processing in this fork is very amateuristic. For more advanced post-processing you can inspect the official repository, which I will probably also do in one of the upcoming months. They will probably release more examples in the near future.
